coursier: init at 1.0.0-M15

adelbertc opened the pull request:
###### Motivation for this change
New package: [coursier](https://github.com/coursier/coursier#command-line) command line

###### Things done

- [x] ~~Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)~~ (Apparently not working for `nix-darwin` at the moment)
- Built on platform(s)
   - [ ] NixOS
   - [X] macOS
   - [ ] Linux
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

adelbertc committed:
coursier: init at 1.0.0-M15

ndowens commented on pkgs/development/tools/coursier/default.nix:
```diff
@@ -0,0 +1,28 @@
+    sha256 = "610c5fc08d0137c5270cefd14623120ab10cd81b9f48e43093893ac8d00484c9";
+  };
+
+  buildInputs = [ makeWrapper ];
```
makeWrapper belongs in nativeBuildInputs

NeQuissimus commented on pkgs/development/tools/coursier/default.nix:
```diff
@@ -0,0 +1,28 @@
+    wrapProgram $out/bin/coursier --prefix PATH ":" ${jre}/bin ;
+  '';
+
+  meta = with stdenv.lib; {
```
Would you like to add yourself as a maintainer?

Either way, can you add me, as I will be using this?

adelbertc committed:
coursier: makeWrapper as nativeBuildInputs, add adelbertc and nequissimus as maintainers

adelbertc commented:
@NeQuissimus So I added the changes suggested above, ran `nix-build -A coursier -K` and it seems to work. I tested this by creating a `in/foo.md` with a tut shed in it and running the following command (taken from [here](https://github.com/tpolecat/tut#quick-start-standalone)):

```
$ ./result/bin/coursier launch -r "https://dl.bintray.com/tpolecat/maven/" org.tpolecat:tut-core_2.11:0.4.8 -- in out '.*\.md$' -classpath $(./result/bin/coursier fetch -p com.chuusai:shapeless_2.11:2.3.1)
```

and I do see it saying "compiling: in/foo.md".

Then out of curiosity I decided to see what happens if I removed `makeWrapper` and the `jre`:

```diff
diff --git a/pkgs/development/tools/coursier/default.nix b/pkgs/development/tools/coursier/default.nix
index 7a2b3e8e1d..17cd828a64 100644
--- a/pkgs/development/tools/coursier/default.nix
+++ b/pkgs/development/tools/coursier/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, makeWrapper, jre }:
+{ stdenv, fetchurl }:

 stdenv.mkDerivation rec {
   name    = "coursier-${version}";
@@ -9,15 +9,12 @@ stdenv.mkDerivation rec {
     sha256 = "610c5fc08d0137c5270cefd14623120ab10cd81b9f48e43093893ac8d00484c9";
   };

-  nativeBuildInputs = [ makeWrapper ];
-
   phases = "installPhase";

   installPhase = ''
     mkdir -p $out/bin
     cp ${src} $out/bin/coursier
     chmod +x $out/bin/coursier
-    wrapProgram $out/bin/coursier --prefix PATH ":" ${jre}/bin ;
   '';

   meta = with stdenv.lib; {
```

I confirmed `./result/bin` now only contains `coursier` instead of both the wrapper and unwrapper version. Yet when I run the same command above it continues to work.. does this mean `tut` doesn't need the JRE? And if not, then where is it coming from? I downloaded the `coursier` script and looked inside and the first thing it does is call `exec java`..

NeQuissimus commented:
I take from your edit that you're good? :D

adelbertc commented:
Yes :-) Patch with suggested changes coming once I get off this train.

adelbertc commented:
Patch pushed!

NeQuissimus commented:
Thanks for the PR!
